src/typeorm-aurora-data-api-driver.ts

```diff
@@ -1,95 +1,113 @@
-          currentQuote = currentCharacter
-        } else if (currentQuote === currentCharacter) {
-          currentQuote = null
-        }
-
-        newQueryString += currentCharacter
-      } else {
-        newQueryString += currentCharacter
       }
+
+      continue
     }
 
-    if (
-      parameters &&
-      parameters.length > 0 &&
-      parameters.length % numberOfParametersInQueryString !== 0
-    ) {
-      throw new Error(`Number of parameters mismatch, got ${numberOfParametersInQueryString} in query string \
-            and ${parameters.length} in input`)
+    if (quoteCharacters.includes(currentCharacter) && !currentCharacterEscaped) {
+      if (!currentQuote) {
+        currentQuote = currentCharacter
+      } else if (currentQuote === currentCharacter) {
+        currentQuote = null
+      }
     }
 
-    const transformedParameters: any[] = []
+    newQueryString += currentCharacter
+  }
+
+  return [newQueryString, numberOfParametersInQueryString]
+}
+
+const transformParameters = (numberOfParametersInQueryString: number, parameters?: any[]) => {
+  if (
+    parameters &&
+    parameters.length > 0 &&
+    parameters.length % numberOfParametersInQueryString !== 0
+  ) {
+    throw new Error(
+      `Number of parameters mismatch, got ${numberOfParametersInQueryString} in query string \
+      and ${parameters.length} in input`)
+  }
 
-    if (parameters && parameters.length > 0) {
-      const numberOfObjects = parameters.length / numberOfParametersInQueryString
+  let extendedParameterCount = 0
+  const transformedParameters: any[] = []
 
-      for (let i = 0; i < numberOfObjects; i += 1) {
-        const parameterObject: any = {}
+  if (parameters && parameters.length > 0) {
+    const numberOfObjects = parameters.length / numberOfParametersInQueryString
 
-        for (let y = 0; y < numberOfParametersInQueryString; y += 1) {
-          const paramName = `param_${y}`
+    for (let i = 0; i < numberOfObjects; i += 1) {
+      const parameterObject: any = {}
 
-          parameterObject[paramName] = parameters[i + y]
-        }
+      for (let y = 0; y < numberOfParametersInQueryString; y += 1) {
+        const parameter = parameters[i + y]
 
-        transformedParameters.push(parameterObject)
+        Array.isArray(parameter)
+          // tslint:disable-next-line: no-increment-decrement
+          ? parameter.forEach(element => parameterObject[`e_param_${extendedParameterCount++}`] = element)
+          : parameterObject[`param_${y}`] = parameter
       }
-    }
 
-    return {
-      queryString: newQueryString,
-      parameters: transformedParameters,
+      transformedParameters.push(parameterObject)
     }
   }
 
-  private readonly region: string
-
-  private readonly secretArn: string
+  return transformedParameters
+}
 
-  private readonly resourceArn: string
+export default class DataApiDriver {
+  public static transformQueryAndParameters(query: string, parameters?: any[]): any {
```

Comment:
Could be a function like these two above rather than a static method

Reply:
I initially did this but had to revert because typeorm had issues dynamically loading the driver module. I initially had them as named exports to be consumed by the test classes directly for better test isolation.

I will however extract the static class and not export it - I should have done that in the first place. That should work fine.